libcontainer/devices/devices_test.go

```diff
@@ -55,9 +65,44 @@ func TestHostDevicesIoutilReadDirDeepFailure(t *testing.T) {
 
 		return []os.FileInfo{fi}, nil
 	}
+	defer func() {
```

Comment:
nit: I feel this should be defined as a named function like `func restoreInjectedFuncs()`